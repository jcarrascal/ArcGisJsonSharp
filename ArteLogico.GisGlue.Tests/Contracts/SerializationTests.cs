using ApprovalTests;
using ApprovalTests.Reporters;
using ArteLogico.GisGlue.Contracts;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using NUnit.Framework;
using System.IO;

namespace ArteLogico.GisGlue.Tests.Contracts
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class SerializationTests
    {
        [Test]
        public void SingleBasemapLayerRoundTripSerialization()
        {
            // Based on a sample given by ArcGIS on
            // http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Single_basemap_layer/02qt00000016000000/
            string expected = File.ReadAllText(@"Contracts\SingleBasemapLayer.json");
            var serializer = new JsonSerializer()
            {
                ContractResolver = new CamelCasePropertyNamesContractResolver(),
                DefaultValueHandling = DefaultValueHandling.Ignore
            };

            string result;
            using (var reader = new StringReader(expected))
            using (var jsonReader = new JsonTextReader(reader))
            using (var writer = new StringWriter())
            using (var jsonWriter = new JsonTextWriter(writer))
            {
                jsonWriter.Formatting = Formatting.Indented;
                var webMap = serializer.Deserialize<WebMap>(jsonReader);
                serializer.Serialize(jsonWriter, webMap);
                result = writer.ToString();
            }

            Approvals.Verify(result);
        }

        [Test]
        public void TwoBasemapLayerRoundTripSerialization()
        {
            // Based on a sample given by ArcGIS on
            // http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Two_basemap_layers/02qt00000015000000/
            string expected = File.ReadAllText(@"Contracts\TwoBasemapLayers.json");
            var serializer = new JsonSerializer()
            {
                ContractResolver = new CamelCasePropertyNamesContractResolver(),
                DefaultValueHandling = DefaultValueHandling.Ignore
            };

            string result;
            using (var reader = new StringReader(expected))
            using (var jsonReader = new JsonTextReader(reader))
            using (var writer = new StringWriter())
            using (var jsonWriter = new JsonTextWriter(writer))
            {
                jsonWriter.Formatting = Formatting.Indented;
                var webMap = serializer.Deserialize<WebMap>(jsonReader);
                serializer.Serialize(jsonWriter, webMap);
                result = writer.ToString();
            }

            Approvals.Verify(result);
        }
    }
}