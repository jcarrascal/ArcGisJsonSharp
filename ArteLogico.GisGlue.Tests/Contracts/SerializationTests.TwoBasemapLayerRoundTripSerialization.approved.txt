{
  "operationalLayers": [
    {
      "url": "http://server.arcgisonline.com/ArcGIS/rest/services/Demographics/USA_Median_Age/MapServer",
      "id": "USA_Median_Age_6919",
      "visibility": true,
      "opacity": 0.56,
      "title": "USA Median Age",
      "itemId": "fce0ca8972ae4268bc4a69443b8d1ef5"
    }
  ],
  "baseMap": {
    "baseMapLayers": [
      {
        "id": "World_Light_Gray_Base_7270",
        "opacity": 1,
        "visibility": true,
        "url": "http://services.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer"
      },
      {
        "id": "World_Light_Gray_Reference_6243",
        "isReference": true,
        "opacity": 1,
        "visibility": true,
        "url": "http://services.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Reference/MapServer"
      }
    ],
    "title": "Light Gray Canvas"
  },
  "version": "1.6"
}